engines/ospray/OSPRayMaterial.cpp

```diff
@@ -118,12 +121,19 @@ OSPTexture2D OSPRayMaterial::_createOSPTexture2D(Texture2DPtr texture)
                  << ": " << texture->getWidth() << "x" << texture->getHeight()
                  << "x" << (int)type << std::endl;
 
-    osp::vec2i texSize{int(texture->getWidth()), int(texture->getHeight())};
-    OSPTexture2D ospTexture =
-        ospNewTexture2D(texSize, type, texture->getRawData(),
-                        OSP_TEXTURE_SHARED_BUFFER);
+    OSPTexture ospTexture = ospNewTexture("texture2d");
+
+    osp::vec2i size{int(texture->getWidth()), int(texture->getHeight())};
 
-    assert(ospTexture);
+    ospSet1i(ospTexture, "type", static_cast<int>(type));
+    ospSet2i(ospTexture, "size", size.x, size.y);
+    const auto texelBytes  = ospray::sizeOf(type);
+    const auto totalTexels = size.x * size.y;
+    const auto totalBytes  = totalTexels * texelBytes;
```

Comment:
add a function to Texture2D to get sizeInBytes()? We should even reuse that for the scene size statistics, but your choice.

Reply:
I can add the function, but I'm unsure about the future of this code anyway. As far as I know the 2D texture was only used for simulation and we are going to change that anyway, aren't we?